### PR TITLE
Reload completion on file changes

### LINKED ISSUE #61: Completion: Dependancy graph and method db should be updated when there import changes
if you have this scenario.
```crystal
# File b.cr
class A
   def self.a
   end
end
```

```crystal
# File a.cr
```

and then change file a.cr to:

```crystal
# File a.cr
require "./b"
A.
```

Completion on A. doesn't work. 
This is because the dependancy to b.cr wasn't added to the file.
We can either change this on save or on update. But I think on update might be really slow.

### PR DESCRIPTION
**Pull request by faustinoaq**
This PR does:

- ~~For some reason the graph is not getting updated yet~~ Fixed! :tada:  see https://github.com/crystal-lang-tools/scry/pull/98#issuecomment-385419208
- Based on #92 (This PR depends on #92)

### Changes

- Reload graph for new file changes given a `text_document.filename`
- Delete completion file node from dependency graph on drop_file
- Fixes #61 

/cc @laginha87

### CODE REVIEW

**Comment by laginha87** [comment]
Damn was also working on fixing this one :P 
Looks like the right way to go 👍

**Comment by faustinoaq** [comment]
@laginha87 Yeah, I;m trying to fix #61, although, this PR doesn't work yet. The graph is not updated on `.rebuild` and I don't know why :sweat_smile: 

Maybe I'm checking the wrong file? on `if file == filename` ?

**Comment by laginha87** [comment]
that bit is a bit overkill you should be able to just run `process_requires` on the filename
Also you need to update the method_db for the open file for it to work. For now a quick test you can do for the dependancy graph part is close and open the file.

**Comment by faustinoaq** [comment]
@crystal-lang-tools/scry Can I get another approval here? :sweat_smile:

File: src/scry/workspace.cr [hunk 1]
```diff
@@ -9,16 +9,26 @@ module Scry
...
 
+    @lookup_path : Array(String)
+
     def initialize(root_uri, process_id, max_number_of_problems)
       @root_uri = root_uri
       @process_id = process_id
       @max_number_of_problems = max_number_of_problems
       @open_files = {} of String => {TextDocument, Completion::MethodDB}
       @dependency_graph = Completion::DependencyGraph::Graph.new
+      @lookup_path = ENV["CRYSTAL_PATH"].split(":") + [@root_uri]
```
Comment (bew) [inline src/scry/workspace.cr]: Small note: Remember: If you're doing `+ [expr]` you're having 2 new arrays!

Comment (faustinoaq) [reply src/scry/workspace.cr]: What about `ENV["CRYSTAL_PATH"].split(":") << @root_uri` ?

**Pull request merged**